src/classes/CloudConnection.ts

```diff
@@ -35,10 +35,13 @@ class CloudConnection {
       }
     });
     this.connection.on("message", (e) => {
+      this.emit("message-may-be-empty", e);
       if (!e) return;
+      this.emit("message", e);
```

Comment:
For situations when the raw responses need to be handled. (e.g. additional metadata sent by custom servers)

Reply:
Custom servers don't have support right now if I remember correctly. That's something I would add in maybe 2.3.0 (this will go 2.2.0)

Reply:
webdev03: A server might send data about user joins/disconnects in some cases.